### PR TITLE
Fix ONNX frontend correctness issues

### PR DESCRIPTION
**Pull request by whjthu**
## 解决的问题

`OnnxStub` 在模型导入、动态输入重新分配和 `to_onnx()` 导出过程中存在若干正确性问题，可能导致调用者模型被修改、节点重复创建、权重数据丢失或导出模型语义错误。

## 主要修改

- 深拷贝传入的 `ModelProto`，避免节点补名、排序和 simplify 修改调用者对象。
- 修复拓扑排序重复处理节点的问题，并补充循环和缺失输入的详细错误信息。
- 统一处理 ONNX 静态输入及 `""` 可选输入占位。
- 修复 Clip、Squeeze、Unsqueeze、Split、Slice、Pad、Dropout、ReduceSum、Reshape 和 Expand 的静态参数读取。
- Pad 仅接受当前后端支持的零值 constant padding，其他模式明确拒绝。
- 推理态 Dropout 映射为 Identity，训练态和 mask 输出明确拒绝。
- 将 ConstantOfShape 从错误的 Constant 路径移除并明确标记为暂不支持。
- 校验 `set_input()` 的输入 shape 数量。
- 动态 shape 重新分配内存后重新复制全部 initializer，避免权重数据丢失。
- 修复 `get_perf_time()` 未返回底层结果的问题。
- 修复 ConvTranspose、Softmax、Split、Expand、Clip 和 LRN 的 ONNX 导出。
- 增加 Softmax axis 的只读 FFI accessor。
- 导出 initializer 前进行深拷贝，保证重复调用 `to_onnx()` 的结果稳定。
- 将依赖外部 ResNet 文件的动态测试替换为内存构造的 MatMul 模型。

## 测试

- CPU ONNX 新增测试：通过，CUDA 专用测试按预期跳过。
- CPU 原有 ONNX 测试：58/58 通过。
- CPU CTest：40/40 通过。
- CUDA ONNX 新增测试：21/21 通过。
- CUDA 原有 ONNX 测试：58/58 通过。
- CUDA CTest：77/77 通过。
- `git diff --check` 和 Ruff 检查通过。

## 影响范围

本 PR 仅修改 ONNX 前端、最小只读 FFI 和相关测试，不新增 operator、kernel 或 runtime，也不包含 InfiniOps、Python InferenceSession 及 `copyout_numpy()` 的独立修复。

### CODE REVIEW

**Comment by whjthu** [comment]
附上逐项改动说明，方便 review

| 序号 | 代码位置 | 修改方式 | 修改原因 | 不修改的影响 |
|---|---|---|---|---|
| 1 | `onnx.py:48` | 进入 `OnnxStub.__init__()` 后深拷贝 `ModelProto`，simplify 使用另一个副本 | 节点补名、排序和 `onnxsim` 可能原地修改模型 | 调用者传入的模型会发生变化，重复加载、保存或比较可能得到不同结果 |
| 2 | `onnx.py:85` | 使用 `pending_nodes` 保存尚未处理的节点 | 旧拓扑排序会在下一轮重新处理已排序节点 | 非拓扑顺序的合法 DAG 可能重复创建同一个算子 |
| 3 | `onnx.py:102` | 排序无法继续时列出节点类型、节点名和缺失输入 | 无法继续不一定是环，也可能是输入缺失 | 所有错误都显示为 `Graph has cycle`，难以定位模型问题 |
| 4 | `onnx.py:1583` | `_parse_attribute()` 默认参数由共享 `dict()` 改为 `None`，传入默认值时先复制 | Python 可变默认参数只创建一次 | 前一个节点的属性可能泄漏到后一个节点 |
| 5 | `onnx.py:1539` | 新增 `_has_input()`，同时检查索引存在且输入名非空 | ONNX 使用 `""` 表示省略的可选输入 | `Clip(["x", "", "max"])` 等合法节点会访问 `data[""]` 并报错 |
| 6 | `onnx.py:1543` | 新增 `_parse_static_input()` 统一读取 initializer 和 Constant | shape、axes、pads 等参数当前需要在构图时确定 | 动态 Tensor 可能被错误当成静态参数，或产生不可读的 `KeyError` |
| 7 | `onnx.py:1568` | 新增 `_parse_static_scalar()` 并校验数据只有一个元素 | Clip 边界、Pad 填充值等参数必须是标量 | 多元素 initializer 可能被错误截断或静默接受 |
| 8 | `onnx.py:639` | Clip 的 min/max 分别通过静态标量接口读取 | min 和 max 都是可选输入，并且可以只提供 max | 省略 min 时导入失败，动态边界错误也无法明确定位 |
| 9 | `onnx.py:672` | Reshape 和 Expand 的 shape 使用 `required=True` 读取 | 当前 GraphHandler 需要在构图时确定目标 shape | shape 缺失或为运行时 Tensor 时可能进入不支持的路径 |
| 10 | `onnx.py:749` | Squeeze、Unsqueeze、Split 优先读取新版输入，同时兼容旧 opset 属性 | 不同 opset 对 axes/split 的存放位置不同 | 新旧模型不能同时兼容，空输入还可能触发 `KeyError` |
| 11 | `onnx.py:856` | Slice 的 starts/ends 设为必填，axes/steps 按空输入规则处理 | Slice 经常使用 `["x", "starts", "ends", "", ""]` | 使用默认 axes/steps 的合法模型无法导入 |
| 12 | `onnx.py:878` | Pad 只接受 constant mode 和零填充值 | 当前后端只实现零值 padding | reflect、edge 或非零 padding 可能被错误执行为零填充，产生错误数值 |
| 13 | `onnx.py:898` | 推理态 Dropout 映射为 Identity，训练态和 mask 输出明确拒绝 | GraphHandler 没有 `dropout()`，而推理态 Dropout 的语义等价于 Identity | 所有 Dropout 节点都会因调用不存在的方法失败，或错误宣称支持训练语义 |
| 14 | `onnx.py:837` | ReduceMean、ReduceSum 同时兼容 axes 属性和第二输入 | PyTorch 新版通常把 axes 导出为输入 Tensor | 新版模型无法导入，空 axes 输入可能产生 `KeyError` |
| 15 | `onnx.py:1097` | ConstantOfShape 从 Constant 分支移除并明确抛出 `NotImplementedError` | ConstantOfShape 的输入是 shape，`value` 只是填充值，不能按 Constant 处理 | 可能把填充值错误当成完整输出，或者构造错误 Tensor |
| 16 | `onnx.py:1122` | 除原有 FUID 映射外，增加 `_initializer_by_name` | FUID 适合导出，名称适合在重新分配后定位 Tensor | 动态 shape 重新分配后无法可靠恢复权重 |
| 17 | `onnx.py:1485` | `init()` 在 `data_malloc()` 后调用 `_copy_initializers()` | 内存重新分配可能改变全部 Tensor 的底层地址 | 动态 batch 后 MatMul、Conv 等权重数据可能丢失、变成零或产生错误结果 |
| 18 | `onnx.py:1502` | `set_input()` 校验 shape 数量与模型输入数一致 | 旧代码使用 `zip()`，长度不一致会静默截断 | 部分输入没有更新 shape，但调用不会报错 |
| 19 | `onnx.py:1511` | shape inference 后调用统一的 `init()` | 输出 shape 变化后需要重新分配内存并恢复 initializer | 只分配内存而不恢复权重会导致后续推理数值错误 |
| 20 | `onnx.py:1530` | `get_perf_time()` 增加 `return` | 原实现调用底层函数后丢弃结果 | Python 调用者永远只能得到 `None` |
| 21 | `onnx.py:1139` | `to_onnx()` 的 Context 状态改为实例属性 | 旧代码使用类属性，不同导出调用共享容器 | 重复导出会累积上一次的节点、名称和 initializer |
| 22 | `onnx.py:1180` | 导出 initializer 前先深拷贝，再修改导出名称 | 导出过程会为 Tensor 重新生成名称 | 直接改名会修改 `OnnxStub` 内部 TensorProto，影响后续导出和执行 |
| 23 | `onnx.py:1263` | ConvTranspose 的 pads 改为四项 `[ph, pw, ph, pw]` | 二维 ONNX ConvTranspose 要求 beginning/end 共四项 padding | 导出模型可能无法通过 ONNX checker，或 padding 语义不完整 |
| 24 | `onnx.py:1339` | Softmax 导出时读取并写入实际 axis | axis 决定归一化维度 | 导出后使用默认 axis，模型可能合法但数值语义错误 |
| 25 | `onnx.py:1389` | Split 根据每个输出 Tensor 的实际 shape 生成 split | 旧实现假设所有输出等长 | `[1, 3]` 等不等长 Split 会被错误导出为 `[2, 2]` |
| 26 | `onnx.py:1433` | Clip 使用合法的 `""` 可选输入，并让边界 dtype 跟随输入 | ONNX 可选输入和边界类型有明确要求 | 可能生成无数据标量 initializer，或出现 FLOAT/FLOAT16 类型不匹配 |
| 27 | `onnx.py:1458` | Expand 的 shape 导出为第二个 INT64 initializer 输入 | 当前 ONNX 规范要求 shape 是输入，不是属性 | 导出模型可能无法通过 ONNX checker |
| 28 | `onnx.py:1466` | LRN 使用 `alpha=...`、`beta=...` 等命名属性 | `make_node()` 的这些值必须作为属性传入 | 位置参数可能进入错误字段，产生异常或非法节点 |
| 29 | `ffi_infinitensor.cc:328` | 新增只读 `softmax_axis_of()` 并注册给 Python | Softmax axis 只保存在 C++ `SoftmaxObj` 中 | Python 导出层无法恢复 Softmax 的真实 axis |
| 30 | `test_onnx.py:597` | 外部 ResNet 动态测试替换为内存构造的 MatMul 模型 | 干净 checkout 中不一定存在 `resnet18-v2-7.onnx` | 代码没有回归也会因为测试资产缺失而失败 |
| 31 | `test_onnxstub.py` | 新增模型不可变、拓扑排序、静态输入、动态内存和导出回归测试 | 原测试未覆盖这些边界场景 | 相同问题后续可能重新出现而无法被 CI 发现 |
| 32 | `Makefile:61` | 将 `test_onnxstub.py` 加入 `make test-onnx` | 新测试需要进入统一测试入口 | 开发者和 CI 运行现有命令时不会执行新增回归测试 |

File: pyinfinitensor/src/pyinfinitensor/onnx.py [hunk 1]
```diff
@@ -849,46 +862,53 @@ def clamp(nums):
...
-                        ),
-                        (
-                            _parse_data(data[node.input[2]])[0]
-                            if len(node.input) > 2
-                            else False
-                        ),
-                    ),
-                ):
-                    tensors[name] = tensor
+                _parse_static_scalar(data, node, 1)
```
Comment (kilinchange) [inline pyinfinitensor/src/pyinfinitensor/onnx.py]: Image
这里不应该强制解析 ratio，因为 onnx 规定在推理模式下 ratio 被忽略，所以即使 ratio 是动态输入，只要 training_mode=False，仍然应该退化为 Identity 操作，而不是直接报错。

File: pyinfinitensor/src/pyinfinitensor/onnx.py
Comment (kilinchange) [inline pyinfinitensor/src/pyinfinitensor/onnx.py]: https://github.com/InfiniTensor/InfiniTensor/blob/f2352911e57febf118556282363b617bb0fbe6dc/src/ffi/ffi_infinitensor.cc#L206
发现 ConvTranspose 这里更大的问题是解包时顺序错了，cpp 侧 backend.conv_trans_attrs_of (op) 当前的返回顺序是 `ph, pw, dh, dw, sh, sw, oph, opw`，但这里解包时把 strides 和 dilations 的顺序交换了。当 strides 与 dilations 不同时，会导致导出的 ONNX 模型里这两个字段是错误的。

**Review by kilinchange — CHANGES_REQUESTED** [review]

**Review by kilinchange — APPROVED** [review]
